classes/PrestaShopAutoload.php

```diff
@@ -286,9 +286,10 @@ protected function getClassesFromDir($path, $hostMode = false)
                 } elseif (substr($file, -4) == '.php') {
                     $content = file_get_contents($rootDir . $path . $file);
 
-                    $namespacePattern = '[\\a-z0-9_]*[\\]';
-                    $pattern = '#\W((abstract\s+)?class|interface)\s+(?P<classname>' . basename($file, '.php') . '(?:Core)?)'
-                                . '(?:\s+extends\s+' . $namespacePattern . '[a-z][a-z0-9_]*)?(?:\s+implements\s+' . $namespacePattern . '[a-z][\\a-z0-9_]*(?:\s*,\s*' . $namespacePattern . '[a-z][\\a-z0-9_]*)*)?\s*\{#i';
+                    $namePattern = '[a-z_\x7f-\xff][a-z0-9_\x7f-\xff]*';
+                    $nameWithNsPattern = '(?:\\\\?(?:' . $namePattern . '\\\\)*' . $namePattern . ')';
```

Comment:
Shouldn't be this?
```suggestion
                    $nameWithNsPattern = '(?:\\\\)?(?:' . $namePattern . '\\\\)*' . $namePattern;
```

Reply:
No, it shouldn't. `\\\\` is properly escaped `\` char and the `(?:` + `)` outside parenthesis are there for safe use with quantifiers (imagine `ab` later used as `ab?` instead of `(?:ab)?`).

Reply:
`?:` only mean `Non-capturing group`.
And `ab?` is different than `(?:ab)?)`.
But here I understand that `\\\\` is only one character, and I think my suggestion is easier to understand, (especially for people not familiar with regex ^^)

Reply:
> `?:` only mean `Non-capturing group`.
> But here I understand that `\\\\` is only one character, and I think my suggestion is easier to understand, (especially for people not familiar with regex ^^)

Yes, I know. If you want, wrap the first `\\\\` inside another `(?:` + `)`, but it is not needed, as it is only one character.

> And `ab?` is different than `(?:ab)?)`.

Yes, that is the example of the difference, beaware that the second/last `)` is part of the text :)

Reply:
> Yes, I know. If you want, wrap the first \\\\ inside another (?: + ), but it is not needed, as it is only one character.

Of course it's not needed, but I think many people are not familiar with regex and if you ask to many people they will answer you it's two `\` :D But indeed, let's merge your PR :+1: